File: training-data/objectivec/AnagramCheck.m
#import <Foundation/Foundation.h>

NSString *normalizeString(NSString *s) {
    NSString *lower = [s.lowercaseString stringByReplacingOccurrencesOfString:@" " withString:@""];
    NSMutableArray *chars = [NSMutableArray array];
    for (NSUInteger i = 0; i < lower.length; i++) {
        [chars addObject:[NSString stringWithFormat:@"%C", [lower characterAtIndex:i]]];
    }
    NSArray *sorted = [chars sortedArrayUsingSelector:@selector(compare:)];
    return [sorted componentsJoinedByString:@""];
}

BOOL isAnagram(NSString *a, NSString *b) {
    return [normalizeString(a) isEqualToString:normalizeString(b)];
}

int main(int argc, const char *argv[]) {
    @autoreleasepool {
        NSLog(@"%d", isAnagram(@"listen", @"silent"));
        NSLog(@"%d", isAnagram(@"hello", @"world"));
    }
    return 0;
}

File: training-data/prolog/anagram_check.pl
normalize_str(Str, Normalized) :-
    string_lower(Str, Lower),
    string_chars(Lower, Chars),
    exclude([C]>>(C == ' '), Chars, Filtered),
    msort(Filtered, Normalized).

is_anagram(A, B) :-
    normalize_str(A, NA),
    normalize_str(B, NB),
    NA == NB.

:- (is_anagram("listen", "silent") -> writeln(true) ; writeln(false)).
:- (is_anagram("hello", "world") -> writeln(true) ; writeln(false)).
